penguin/tools/providers/mcp.py

```
@@ -0,0 +1,125 @@
+        if not self.enabled:
+            return json.dumps(
+                {
+                    "error": "mcp_disabled",
+                    "tool": tool_name,
+                    "message": "MCP is not enabled in Penguin config.",
+                }
+            )
+        try:
+            result = self.manager.call_tool_sync(tool_name, tool_input or {})
+            return json.dumps({"status": "ok", "result": result}, indent=2)
+        except Exception as exc:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)